File: tiny-pro/tinypro-model/src/main/java/com/huaweicloud/model/Menu.java
package com.huaweicloud.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.Getter;

@Data
@Getter
@Entity(name="menu")
public class Menu {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer id;
    @Column()
    String name;
    @Column(name = "`order`")
    Integer order;
    @Column(nullable = true)
    Integer parentId;
    @Column()
    String menuType;
    @Column(nullable = true)
    String icon;
    @Column()
    String component;
    @Column()
    String path;
}

File: egg/app/models/menu.ts
import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';

@Entity('menu')
export class Menu {
  @PrimaryGeneratedColumn()
  id: number;
  @Column()
  name: string;
  @Column()
  order: number;
  @Column({ nullable: true })
  parentId: number | null;
  @Column()
  menuType: string;
  @Column({ nullable: true })
  icon: string | null;
  @Column()
  component: string;
  @Column()
  path: string;
}
